feature_engine/discretisation/gemoetric_width.py

```diff
@@ -0,0 +1,152 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._docstrings.fit_attributes import (
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.init_parameters import _variables_numerical_docstring
+from feature_engine._docstrings.methods import _fit_transform_docstring
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine._variable_handling.init_parameter_checks import (
+    _check_init_parameter_variables,
+)
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+
+
+@Substitution(
+    return_object=BaseDiscretiser._return_object_docstring,
+    return_boundaries=BaseDiscretiser._return_boundaries_docstring,
+    binner_dict_=BaseDiscretiser._binner_dict_docstring,
+    fit=BaseDiscretiser._fit_docstring,
+    transform=BaseDiscretiser._transform_docstring,
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+    power="{1/n}",
+    subindex="{i+1}"
+)
+class GeometricWidthDiscretiser(BaseDiscretiser):
+    """
+    The `GeometricWidthDiscretiser()` divides continuous numerical variables into
+    intervals of increasing width. The width of each succeeding interval is larger than
+    the previous interval by a constant amount (cw).
+
+    The constant amount is calculated as:
+
+    .. math::
+        cw = (Max - Min)^{power}
+
+    were Max and Min are the variable's maximum and minimum value, and n is the number
+    of intervals.
+
+    The sizes of the intervals themselves are calculated with a geometric progression:
+
+    .. math::
+        a_{subindex} = a_i cw
+
+    Thus, the first interval's width equals cw, the second interval's width equals
+    2 * cw, and so on.
+
+    The `GeometricWidthDiscretiser()` works only with numerical variables. A list of
+    variables can be passed as argument. Alternatively, the discretiser will
+    automatically select all numerical variables.
+
+    The `GeometricWidthDiscretiser()` first finds the boundaries for the intervals for
+    each variable. Then, it transforms the variables, that is, sorts the values into
+    the intervals.
+
+    More details in the :ref:`User Guide <geometric_width_discretiser>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    bins: int, default=10
+        Desired number of intervals / bins.
+
+    {return_object}
+
+    {return_boundaries}
+
```

Comment:
we need to add precision here, and in all discretizers

Reply:
its private. We need the _ so that the user does not see it when pressing tab

Reply:
I think sklearn does not support private parameters in init